// nudity-filter/sightengine.js
var API_ENDPOINT = "http://api.sightengine.com/";
var API_VERSION = "1.0";

var path = require('path');
var request = require('request');
var fs = require('node-fs');

Sightengine = function(api_user, api_secret) {
	this.api_user = api_user;
	this.api_secret = api_secret;
}

Sightengine.prototype.checkNudityForURL = function(url, callback) {
	request(API_ENDPOINT+'/'+API_VERSION+'/nudity.json?api_user='+this.api_user+'&api_secret='+this.api_secret+'&url='+url, function(error, response, body) {
		requestCallback(error,response,body,callback);
	});
}

Sightengine.prototype.checkNudityForFile = function(filepath, callback) {
	this.callback = callback;
	var formData = {
		api_user: this.api_user,
		api_secret: this.api_secret,
		photo: {
			value:  fs.createReadStream(filepath),
			options: {
				filename: path.basename(filepath)
			}

		}
	};

	request.post({url:API_ENDPOINT+'/'+API_VERSION+'/nudity.json', formData: formData}, function(error, response, body) {
		requestCallback(error, response, body, callback);
	});
}

var requestCallback = function(error, response, body, callback) {
                if(error) {
                        callback(error, null);
                }
                else if(response.statusCode>=500) {
                        callback(new Error("Server error "+response.statusCode), null);
                }
                else {
                        var output = JSON.parse(body);
                        if(output.status=="success") callback(null, output.nudity);
                        else callback(new Error(output.error_message), null);
                }
}

module.exports = Sightengine;
